src/main/java/com/amplitude/HttpTransport.java

```diff
@@ -191,6 +197,12 @@ private void retryEventsOnLoop(String userId, String deviceId) {
   private EventsRetryResult retryEventsOnce(String userId, String deviceId, List<Event> events)
       throws AmplitudeInvalidAPIKeyException {
     Response response = httpCall.makeRequest(events);
+      if (logger.getLogMode() == AmplitudeLog.LogMode.DEBUG) {
+          StringBuilder debugMessage = new StringBuilder();
+          debugMessage.append("Retry for userId ").append(userId).append(", deviceId ").append(deviceId)
```

Comment:
Could we just use string format instead here? That will look cleaner.

Reply:
Sure, since concatenating all in one place, there's no performance difference.